contracts/staking/PrtStakingPool.sol

```diff
@@ -0,0 +1,351 @@
+/*
+    Copyright 2024 Index Cooperative
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    SPDX-License-Identifier: Apache License, Version 2.0
+*/
+
+pragma solidity ^0.6.10;
```

Comment:
I originally opted for this version of `openzeppelin-contracts` because it [still has ERC20Snapshot](https://forum.openzeppelin.com/t/erc20snapshot-alternative-in-new-oz-library/38137/1) (and convenience with the repo)

However we may want to reconsider for the [EIP712 functionality](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/cryptography/EIP712.sol) in later versions